tests/providers/amazon/aws/transfers/test_s3_to_redshift.py

```diff
@@ -65,6 +65,45 @@ def test_execute(self, mock_run, mock_session):
         assert secret_key in copy_query
         assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], copy_query)
 
+    @mock.patch("boto3.session.Session")
+    @mock.patch("airflow.providers.postgres.hooks.postgres.PostgresHook.run")
+    def test_execute_with_column_list(self, mock_run, mock_session):
+        access_key = "aws_access_key_id"
+        secret_key = "aws_secret_access_key"
+        mock_session.return_value = Session(access_key, secret_key)
+        mock_session.return_value.access_key = access_key
+        mock_session.return_value.secret_key = secret_key
+        mock_session.return_value.token = None
+
+        schema = "schema"
+        table = "table"
+        s3_bucket = "bucket"
+        s3_key = "key"
+        column_list = ["column_1", "column_2"]
+        copy_options = ""
+
+        op = S3ToRedshiftOperator(
+            schema=schema,
+            table=table,
+            s3_bucket=s3_bucket,
+            s3_key=s3_key,
+            column_list=column_list,
+            copy_options=copy_options,
+            redshift_conn_id="redshift_conn_id",
+            aws_conn_id="aws_conn_id",
+            task_id="task_id",
+            dag=None,
+        )
+        op.execute(None)
+
+        credentials_block = build_credentials_block(mock_session.return_value)
+        copy_query = op._build_copy_query(credentials_block, copy_options)
```

Comment:
Can you copy the result of this function to the test?

Reply:
Hi @mik-laj :) Thanks for the review.

Could you explain it in more detail? 

Do you want the result of `op._build_copy_query` to be asserted with expected plain copy text?
I thought that line 105 ` assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], copy_query)` checks the result


FYI expected `copy_query` looks like this 
```sql
COPY schema.table (column_1, column_2)
FROM 's3://bucket/key'
with credentials
'aws_access_key_id=aws_access_key_id;aws_secret_access_key=aws_secret_access_key'
;
```

Reply:
> FYI expected copy_query looks like this

Can you add it to tests?

Reply:
Oh then like this?

```suggestion
        copy_query = op._build_copy_query(credentials_block, copy_options)
        expected_copy_query = '''
                              COPY schema.table (column_1, column_2)
                              FROM 's3://bucket/key'
                              with credentials
                              'aws_access_key_id=aws_access_key_id;aws_secret_access_key=aws_secret_access_key'
                              ;
                              '''
        assert_equal_ignore_multiple_spaces(self, expected_copy_query, copy_query)

```

Reply:
Exactly. Building this SQL is crucial in integrating with a thirty-patty service, so it should show up in tests.

Reply:
Can you add tests for SQL without columns?

Reply:
Implemented !

SQL without columns
https://github.com/apache/airflow/blob/d1dc18d9c84c2b3d1f3005a69f93a1adc59476b8/tests/providers/amazon/aws/transfers/test_s3_to_redshift.py#L62-L69


SQL with columns
https://github.com/apache/airflow/blob/d1dc18d9c84c2b3d1f3005a69f93a1adc59476b8/tests/providers/amazon/aws/transfers/test_s3_to_redshift.py#L109-L116